src/coreclr/vm/interpexec.cpp

```diff
@@ -378,7 +378,11 @@ typedef void (*HELPER_FTN_V_PP)(void*, void*);
 InterpThreadContext::InterpThreadContext()
 {
     // FIXME VirtualAlloc/mmap with INTERP_STACK_ALIGNMENT alignment
+#ifdef TARGET_UNIX
+    pStackStart = pStackPointer = (int8_t*)aligned_alloc(INTERP_STACK_ALIGNMENT, INTERP_STACK_SIZE);
```

Comment:
Coreclr itself doesn't call Unix specific functions. This should be hidden in the minipal. Historically, PAL was used for this stuff and we are trying to reduce it. The minipal is the new place for abstracting platform dependencies. 

Can you please add a new method for aligned allocation to the VMToOSInterface in https://github.com/dotnet/runtime/blob/3345865f57f59f4a54e1749c06f5a2782d961a16/src/coreclr/minipal/minipal.h and implement it in files in the related Unix and Windows subfolders? I would recommend adding a new source file named "memory.cpp" there. In the future, I would add support for memory mapping functionality there and remove that from the PAL.

Also, this would not work on macOS, it should use `posix_memalign` there instead.